test/src/main/java/org/apache/accumulo/test/fate/FateInterleavingIT.java

```diff
@@ -254,35 +253,37 @@ protected void testInterleaving(FateStore<FilTestEnv> store, ServerContext sctx)
       // thread to interleave and work on another fate id, but may not always interleave.
       // It is unlikely that the FATE will not interleave at least once in a run, so we will check
       // for at least one occurrence.
-      int interleaves = 0;
-      int i = 0;
-      Map.Entry<FateId,String> prevOp = null;
+      final AtomicInteger interleaves = new AtomicInteger();
+      final AtomicInteger i = new AtomicInteger();
+      final AtomicReference<Map.Entry<FateId,String>> prevOp = new AtomicReference<>(null);
       var expRunOrder = List.of("FirstOp::isReady1", "FirstOp::isReady2", "FirstOp::call",
           "SecondOp::isReady1", "SecondOp::isReady2", "SecondOp::call", "LastOp::isReady1",
           "LastOp::isReady2", "LastOp::call");
       var fateIdsToExpRunOrder = Map.of(fateIds[0], new ArrayList<>(expRunOrder), fateIds[1],
           new ArrayList<>(expRunOrder), fateIds[2], new ArrayList<>(expRunOrder));
 
-      while (iter.hasNext()) {
-        var currOp = iter.next();
-        FateId fateId = currOp.getKey();
-        String currStep = currOp.getValue();
-        var expRunOrderFateId = fateIdsToExpRunOrder.get(fateId);
-
-        // An interleave occurred if we do not see <FateIdX, OpN::isReady2> immediately after
-        // <FateIdX, OpN::isReady1>
-        if (prevOp != null && prevOp.getValue().contains("isReady1")
-            && !currOp.equals(new AbstractMap.SimpleImmutableEntry<>(prevOp.getKey(),
-                prevOp.getValue().replace('1', '2')))) {
-          interleaves++;
-        }
-        assertEquals(currStep, expRunOrderFateId.remove(0));
-        prevOp = currOp;
-        i++;
+      try (Scanner scanner = client.createScanner(FATE_TRACKING_TABLE)) {
+        scanner.stream().map(FateInterleavingIT::toIdStep).forEach(currOp -> {
```

Comment:
Used the stream directly without the need for converting an iterator. Has the cost of needing to make the variables set within the loop final and atomic.

Reply:
I think an iterator is better for this so we can avoid using atomic.
I should probably close the scanner(s) though